cate/ds/esa_cci_odp.py

```diff
@@ -1441,7 +1441,8 @@ async def _init_file_list(self, monitor: Monitor = Monitor.NONE):
         # Compute the data source's temporal coverage
         for file_rec in file_list:
             if file_rec[1]:
-                file_start_date = datetime.strptime(file_rec[1].split('.')[0], _TIMESTAMP_FORMAT)
+                file_start_date = datetime.strptime(file_rec[1].split('/')[0].split('.')[0].split('+')[0],
```

Comment:
The intention of this expression is barely understandable. Leave a comment here to document what's going on ie why this is needed.